File: bench/nqueens.php
<?php
$count = 0;
function check($board, $row, $col) {
    for ($i = 0; $i < $row; $i++)
        if ($board[$i] == $col || $board[$i] - $i == $col - $row || $board[$i] + $i == $col + $row)
            return 0;
    return 1;
}
function solve(&$board, $row, $n) {
    global $count;
    if ($row == $n) { $count++; return; }
    for ($col = 0; $col < $n; $col++)
        if (check($board, $row, $col)) {
            $board[$row] = $col;
            solve($board, $row + 1, $n);
        }
}
$board = array_fill(0, 14, 0);
solve($board, 0, 13);
echo "$count\n";

File: bench/nqueens.py
def check(board, row, col):
    for i in range(row):
        if board[i] == col or board[i] - i == col - row or board[i] + i == col + row:
            return False
    return True

def solve(board, row, n):
    global count
    if row == n:
        count += 1
        return
    for col in range(n):
        if check(board, row, col):
            board[row] = col
            solve(board, row + 1, n)

count = 0
solve([0]*14, 0, 13)
print(count)
